src/compliment/sources/class_members.clj

```diff
@@ -114,21 +129,27 @@
 (defn members-candidates
   "Returns a list of Java non-static fields and methods candidates."
   [prefix ns context]
-  (when (class-member-symbol? prefix)
-    (let [prefix (subs prefix 1)
-          inparts? (re-find #"[A-Z]" prefix)
-          klass ^{:lite nil} (try-get-object-class ns context)]
-      (for [[member-name members] (get-all-members ns klass)
-            :when (and (if inparts?
-                         (camel-case-matches? prefix member-name)
-                         (.startsWith ^String member-name prefix))
-                       ^{:lite true}
-                       (or (not klass)
-                           (some (fn [m] (.isAssignableFrom (.getDeclaringClass ^Member m) klass))
-                                 members)))]
-        {:candidate (str "." member-name)
-         :type (if (instance? Method (first members))
-                 :method :field)}))))
+  (when-let [[klass-name prefix] (class-member-symbol? prefix)]
+    (let [prefix      (str/replace prefix #"^\." "")
+          qualified?  (not-empty klass-name)
+          inparts?    (re-find #"[A-Z]" prefix)
+          klass       (if qualified?
+                        (resolve-class ns (symbol klass-name))
+                        ^{:lite nil} (try-get-object-class ns context))
+          all-members (when (or klass (not qualified?))
+                        (get-all-members ns klass))]
+      (for [[member-name members] all-members
+            :when                 (and (if inparts?
+                                         (camel-case-matches? prefix member-name)
+                                         (.startsWith ^String member-name prefix))
+                                       ^{:lite true}
+                                       (or (not klass)
+                                           (some (fn [m] (.isAssignableFrom (.getDeclaringClass ^Member m) klass))
+                                                 members)))]
+        {:candidate (cond->> (str "." member-name)
+                      (and qualified? klass) (str klass-name "/"))
+         :type      (if (instance? Method (first members))
+                      :method :field)}))))
```

Comment:
Aligning looks very uglily in this function. It's alright in the first `let`, but later it's just a waste of space.